src/resolve/metadataResolver.ts

```diff
@@ -76,7 +76,20 @@ export class MetadataResolver {
       return components;
     }
 
-    for (const fsPath of this.tree.readDirectory(dir).map(fnJoin(dir))) {
+    for (const fsPath of this.tree
+      .readDirectory(dir)
+      .map(fnJoin(dir))
+      // this method isn't truly recursive, we need to sort directories before files so we look as far down as possible
+      // before finding the parent and returning only it - by sorting, we make it as recursive as possible
+      .sort((a, b) => {
+        if (this.tree.isDirectory(a) && this.tree.isDirectory(b)) {
+          return 0;
+        } else if (this.tree.isDirectory(a) && !this.tree.isDirectory(b)) {
+          return -1;
+        } else {
+          return 1;
+        }
+      })) {
```

Comment:
What do you think about making a sort function and referencing it here rather than all inline?  Or maybe build the array in the line above the for loop?  imho the readability decreased with everything inline.